src/webgpu/api/operation/sampling/derivative.spec.ts

```diff
@@ -0,0 +1,216 @@
+                var r : f32 = 0.0;
+                for (var i = 0; i < uniforms.numIterations; i = i + 1) {
+                  
+                  // Bogus condition to make this a "loop with varying iteration".
+                  if (fragUV.x > 500.0) {
+                    break;
+                  }
+                  
+                  // Do the gradient operations within the loop
+                  let dx = dpdxCoarse(fragUV.x);
+                  let dy = dpdyCoarse(fragUV.y);
+                    r = r + dx + dy;
+                }
+                return vec4<f32>(r);
+            }
+            `,
+        }),
+        entryPoint: 'main',
+        targets: [{ format: kColorAttachmentFormat }],
+      },
+      primitive: { topology: 'triangle-list' },
+    });
+  }
+
+  // return the render target texture object
+  drawQuad(numIterations: number): GPUTexture {
+    // make sure it's already initialized
+    assert(this.pipeline !== undefined);
+
+    const uniformBufferSize = 4; // numIterations : i32
+
+    const uniformBuffer = this.device.createBuffer({
+      size: uniformBufferSize,
+      usage: GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST,
+    });
+
+    const uniforms = Int32Array.from([numIterations]);
+    this.queue.writeBuffer(
+      uniformBuffer,
+      0,
+      uniforms.buffer,
+      uniforms.byteOffset,
+      uniforms.byteLength
+    );
+
+    const bindGroup = this.device.createBindGroup({
+      entries: [{ binding: 0, resource: { buffer: uniformBuffer } }],
+      layout: this.pipeline.getBindGroupLayout(0),
+    });
+
+    const colorAttachment = this.device.createTexture({
+      format: kColorAttachmentFormat,
+      size: { width: kRTSize, height: kRTSize, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: colorAttachmentView,
+          storeOp: 'store',
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+    pass.setPipeline(this.pipeline);
+    pass.setBindGroup(0, bindGroup);
+    pass.draw(6);
+    pass.endPass();
+    this.device.queue.submit([encoder.finish()]);
+
+    return colorAttachment;
+  }
+}
+
+export const g = makeTestGroup(DerivativesTest);
+
+g.test('derivative_in_varying_iteration_loop')
+  .desc(
+    `
+    Derivative test that invokes dpdx/dpdy in a loop where the compiler cannot determine the
+    number of iterations at compile time. FXC typically unrolls such loops and fails with a
+    timeout. See https://crbug.com/tint/1112.
+    `
+  )
+  .fn(async t => {
+    const byteLength = kRTSize * kBytesPerRow;
+    const results = [];
+
+    const allNumIterations = [1, 2, 3];
+
+    for (let i = 0; i < allNumIterations.length; ++i) {
```

Comment:
Can be reduced to:

```ts
    for (const numIterations of allNumIterations) {
      results.push(
        await t.readGPUBufferRangeTyped(t.copyRenderTargetToBuffer(t.drawQuad(numIterations)), {
          type: Float32Array,
          typedLength: byteLength / 4,
        })
      );
    }
```

Reply:
Indeed, was just trying to keep this consistent with the loop below.

Reply:
Done